hibernate-core/src/main/java/org/hibernate/dialect/H2Dialect.java

```diff
@@ -122,7 +122,7 @@ public H2Dialect() {
 		registerColumnType( Types.BIT, "boolean" );
 		registerColumnType( Types.CHAR, "char($l)" );
 		registerColumnType( Types.DATE, "date" );
-		registerColumnType( Types.DECIMAL, buildId >= 201 ? "numeric($p,$s)" : "decimal($p,$s)" );
+		registerColumnType( Types.DECIMAL, "decimal($p,$s)" );
 		registerColumnType( Types.NUMERIC, buildId >= 201 ? "numeric($p,$s)" : "decimal($p,$s)" );
 		registerColumnType( Types.DOUBLE, "double" );
 		registerColumnType( Types.FLOAT, "float" );
```

Comment:
btw, there is a related ticket for the above `float' issue: https://github.com/hibernate/hibernate-orm/pull/2855

Reply:
`AbstractSchemaValidator.validateColumnType()` is the culprit here.

Reply:
And `Table.validateColumns()`, I suppose, except that that method doesn't seem to be called anywhere.